plugin/semantic.py

```diff
@@ -0,0 +1,232 @@
+from .code_actions import CodeActionOrCommand
+from .core.protocol import Diagnostic
+from .core.protocol import Request
+from .core.registry import LspTextCommand
+from .core.registry import windows
+from .core.sessions import SessionBufferProtocol
+from .core.typing import List, Optional, Any, Dict, Tuple, Sequence, cast
+from .core.views import text_document_identifier
+
+import sublime
+import sublime_plugin
+import time
+
+SUBLIME_WORD_MASK = 515
+
+NON_MODIFIER_SCOPES = {
+            'variable'     :'variable.other.lsp',
+            'parameter'    :'variable.parameter.lsp',
+            'function'     :'variable.function.lsp',
+            'method'       :'variable.function.lsp',
+            'property'     :'variable.other.member.lsp',
+            'class'        :'entity.name.class.lsp',
+            'enum'         :'variable.enum.lsp',
+            'enumMember'   :'entity.name.enum.lsp',
+            'type'         :'storage.type.lsp',
+            'macro'        :'variable.other.constant.lsp',
+            'namespace'    :'variable.other.namespace.lsp',
+            'typeParameter':'variable.parameter.generic.lsp',
+            'comment'      :'comment.block.documentation.lsp',
+            'dependent'    :'',
+            'concept'      :'',
+            'module'       :'',
+            'magicFunction':'',
+            'selfParameter':'',
```

Comment:
I think here in this LSP package we should only provide scopes for the standardized token types from the specs. And then we should provide a way for LSP-* packages to add mappings with such custom types only used by a certain server via `AbstractPlugin.configuration`. Those should be merged together then, also allowing LSP-* packages to override scopes for the predefined types, if needed.